### ISSUE TITLE
Domogik Cloud

### ISSUE THREAD

**Issue by Cereal2nd** [domogik/domogik#84]
First, don't worry :)

This ticket is just here to explain what we would like to do...

First, Domogik will NEVER NEED A CLOUD TO WORK.

So, why a cloud ?
Well, it not really a cloud... more an online website which will allow us to do some stats...
## 0. uuid

First, let's assume that each Domogik has an uuid (unique id). This id won't be used for some commercial or any other crazy/dark operations.

Then, let's assume that the choice to send data to a "cloud" will be let to the user.
## 1. System stats

We (Domogik team) would like to get some stats about :

```
the number of living domogik of a release (example : we have 2345 running domogik, 50% are in 0.4.0, 30% in 0.4.1, 20% in dev). This will be really usefull for us to see when to stop maintaining the releases
the plugins releases (in the same way)
domoweb, domodroid, ... releases
```
## 2. A community map for some data

Thanks to a plugin, we could let the user send data about outside to the 'cloud' :

```
gps data or localisation (yes, to create a map we need to know where is the data from...)
temperature
humidity
uv
....
```

Thanks to this, we could one day create a map, and maybe also have a plugin to use your neighbour data from outside! This could help people living not far from each other to help themselves.

Finally... you see, there is nothing to worry about ;)

**Comment by Cereal2nd** [comment]
This system we could reuse to get some error stats inside domogik,
for example if the xplgw queue is to full we can generate an error

The idea would be to generate a mq stats message and manager keeps these in memory, we can then create plugins that monitors these values to generate sms/email/... to notify the admin

Later we can use these values to send anonymus to the domogik team to get some stats

**Comment by ferllings** [comment]
We discussed the very same idea, with fritz_smh, few months ago.
A community plugin, to get some stats as you described.

It's not mandatory (plugin) and anonymous (uuid)

We just need to find a good way incite people to install it.

**Comment by Cereal2nd** [comment]
implemented in 0.5
